hbase-server/src/main/java/org/apache/hadoop/hbase/wal/AbstractRecoveredEditsOutputSink.java

```diff
@@ -111,6 +128,13 @@ protected Path closeRecoveredEditsWriterAndFinalizeEdits(RecoveredEditsWriter ed
       // TestHLogSplit#testThreading is an example.
```

Comment:
Not your fault, but we'd better change the test implementation instead of adding more checks in the normal code path, since this will lead to a request to NN...

Can be a separated issue.